### DISCUSSION TITLE
Unable to import sty files

### DISCUSSION THREAD

**Discussion by csrmy** [Witiko/markdown#489] [Q&A] [answered]
Hi,

I am new to Tex and markdown.
Sorry in advance if this should be a discussions instead of an issue.

sty files are not found with `import` or `theme` option when importing `markdown` package.
The compilation goes through without other issues.

It's not only the `markdownthemewitiko_dot.sty` theme. Originally I noticed when one custom theme from me was not working.
It works with `witiko/markdown/defaults` is it because the `markdownthemewitiko_markdown_defaults.tex` is found ?

I feel like when importing it searches for the Tex file only. 

Tex file:
```
\documentclass[a4paper, 12pt]{book}
\usepackage[import=witiko/dot]{markdown}

\markdownSetup{%
	pipeTables=true,
}

\begin{document}
\begin{markdown}

| 1 | 2 | 3 | 4 |
| :--- | ---- | ---- | ---- |
| 11 | 12 | 13 | 14 |
| 21 | 22 | 23 | 24 |
| 31 | 32 | 33 | 34 |

\end{markdown}
\end{document}
```

Error log:
```
! LaTeX Error: File 'markdownthemewitiko_dot' not found.
```

Nomenclature:
```
├── markdown/
│   ├── markdown.sty
│   ├── markdownthemewitiko_dot.sty
│   └── ...
└── test.tex
```

Logs when importing `witiko/markdown/defaults` :
```
(/usr/share/texmf-dist/tex/generic/markdown/markdownthemewitiko_markdown_defaults.tex))
(/usr/share/texmf-dist/tex/latex/markdown/markdownthemewitiko_markdown_defaults.sty

Package markdown Warning: Plain TeX Markdown theme witiko/markdown/defaults
(markdown)                was previously loaded on line 142, not loading it again
```

**Comment by Witiko** [comment]
Hi again, @l0th3r!

Only the file `markdown.tex` is expected to be placed in the directory `markdown/`. It is a special case, so that users can name their main document `markdown.tex` without any conflicts with the Markdown package, as discussed with @mmarras in https://github.com/Witiko/markdown/issues/74.

You should either:
1. Place the themes into the same directory as your main document.
2. Instruct LaTeX to look for package files in the directory `markdown/` by placing the following code before `\usepackage{markdown}`:

``` tex
\ExplSyntaxOn
\seq_put_right:Nn
  \l_file_search_path_seq
  { ./markdown/ }
\ExplSyntaxOff
```

**Comment by csrmy** [comment]
Thank you for your answer @Witiko !

I have tried to put files like this before writing the issue.

The same situation happens with this nomenclature:
```
├── markdown/
│   └── markdown.tex
├── markdown.sty
├── markdownthemewitiko_dot.sty
├── test.tex
└── ...
```

The file is still not found event tho is located in the same place as my Tex file.

![image](https://github.com/user-attachments/assets/ecf4ad73-d529-4444-9df6-4a287f7aa6be)

**Reply by Witiko** [reply, marked as answer]
Oh, this seems to be a bug from https://github.com/Witiko/markdown/issues/471. The solution is to use `import=witiko/dot` in the `\markdownSetup` command, not in `\usepackage{markdown}` until the bug has been fixed.

In your document, you would write:

``` tex
\documentclass[a4paper, 12pt]{book}
\usepackage{markdown}

\markdownSetup
  {
    pipeTables = true,
    import = witiko/dot,
  }

\begin{document}
\begin{markdown}

| 1 | 2 | 3 | 4 |
| :--- | ---- | ---- | ---- |
| 11 | 12 | 13 | 14 |
| 21 | 22 | 23 | 24 |
| 31 | 32 | 33 | 34 |

\end{markdown}
\end{document}
```

**Reply by csrmy** [reply]
Thank you for you very precious help and time ! Have a good one :)
